### PR TITLE
refactor: GitHub repo cleanup and edited some mod files

### LINKED ISSUE #318: Chore: Add getting started guide to Readme
At the moment we do not provide any information on how to get started with a local repository.

### PR DESCRIPTION
**Pull request by bconlon1**
```
Moved .md files to docs/, cleaned up our README to be more up-to-date and less cluttered, improved the CONTRIBUTING file and added a separate AGREEMENT file for CLA, set up a git hook for conventional commits, and also updated our mods.toml and pack.mcmeta

closes #318
```

### CODE REVIEW

**Comment by bconlon1** [comment]
test

**Comment by bconlon1** [comment]
test

**Comment by bconlon1** [comment]
test

**Comment by bconlon1** [comment]
test

**Review by bconlon1 — COMMENTED** [review]
test

**Review by bconlon1 — COMMENTED** [review]
test

**Review by bconlon1 — COMMENTED** [review]
test

**Comment by bconlon1** [comment]
Related PR: https://github.com/Gilded-Games/.github/pull/1

File: src/main/resources/pack.mcmeta [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
     "pack": {
         "description": "Aether Resources",
-        "pack_format": 10
+        "pack_format": 10,
+        "forge:resource_pack_format": 10,
```
Comment (Burning-Cactus) [inline src/main/resources/pack.mcmeta]: The current resource pack format is 9, data pack format is 10.

Comment (bconlon1) [reply src/main/resources/pack.mcmeta]: Hmm, I must've gotten something mixed up when writing this.

**Pull request merged**